src/GPLVM/PCA.hs

```diff
@@ -0,0 +1,39 @@
+module GPLVM.PCA where
+
+import Universum hiding (Vector)
+
+import GPLVM.Types
+
+import Control.Monad.Identity
+import Data.Array.Repa hiding (Vector)
+import Data.Array.Repa.Repr.ForeignPtr
+import Data.Bifunctor
+import Numeric.LinearAlgebra.Repa hiding (Matrix, Vector)
+import Numeric.LinearAlgebra.Repa.Conversion
+import qualified Numeric.Matrix as M
+
+data PCA = PCA
+  {   inputData :: Matrix D Double
+    , covariance  :: Covariance Double
+    , eigenVectors :: EigenVectors Double
+    , eigenValues :: Vector F Double
+    , finalData :: Matrix D Dbouble
```

Comment:
`Dbouble`? what's that? Some new numerical data type that widely used for PCA? :peka: